packages/grafana-ui/src/components/RefreshPicker/RefreshPicker.tsx

```diff
@@ -93,7 +99,17 @@ export function intervalsToOptions({ intervals = defaultIntervals }: { intervals
   SelectableValue<string>
 > {
   const intervalsOrDefault = intervals || defaultIntervals;
-  const options = intervalsOrDefault.map((interval) => ({ label: interval, value: interval }));
+  const options = intervalsOrDefault.map((interval) => {
+    const label: { [key: string]: string | number } = parseDuration(interval);
+    const unit = Object.keys(label);
```

Comment:
this typing feels a little funky. looking at the `Duration` type in `date-fns`, the RHS of the object is only ever a `number`, not `string | number`. maybe something like:
```suggestion
    const duration = parseDuration(interval);
    const unit = Object.keys(duration) as (keyof Duration)[];
```
would be better. that also suggests that `duration` _could_ be an empty object (`{}`) and therefore `unit` (and `amount`) would be `undefined` and we need to handle that case 🤔 

pinging @joshhunt to check my typescript

Reply:
Good Point @ashharrison90 
The downside with this approach is that typescript won't identify `slice` as a valid string property when used on an array item which belongs to an array of strings(`unit` in this case)

e.g `unit[0]`

Reply:
@thisisobate don't quite follow. `.slice` seems to work fine, the only TS error is about `amount` potentially being undefined:

![image](https://user-images.githubusercontent.com/20999846/140753706-4252bcf3-3f29-46de-9a54-49a24c038a87.png)

Reply:
It seems like it's an issue with my VScode instance...would fix it up. Thanks!

Reply:
We've got an alternate implementation to skip this problem entirely, but imho I think the type was fine before.

If we want to treat the `label`/`duration` as a bag of string: number (as opposed to an object with specific keys), then `Record<string, number>` is the right type for that.

As an absolute last resort, if we must, you could do the cast with `Object.keys(duration) as Array<keyof valueof duration>` so at least there's a higher chance of the cast being correct as the code is changed around. But IMHO `as` type assertions should be avoided at all costs.

Reply:
Then I think it makes sense for us to revert back to the type we were using before. Would push some changes right away

Reply:
If you're using `formatDuration` from date-fns you shouldn't need this type any more.